cloudinit/config/cc_mounts.py

```diff
@@ -223,6 +223,63 @@ def suggested_swapsize(memsize=None, maxsize=None, fsys=None):
     return size
 
 
+def get_fstype(path):
+    """Return the type of the filesystem that path is on.
+    If path is a mount point itself, then the type of the filesystem mounted
+    on path rather than its parent dir."""
+    return util.get_mount_info(path)[1]
+
+
+def create_swapfile(fname, size):
+    """Size is in MB."""
+
+    def unlink_f(fname):
+        if os.path.exists(fname):
+            os.unlink(fname)
+
+    def create_swap(fname, size, method):
+        errmsg = "Failed to create swapfile '%s' of size %dMB via %s: %s"
+
+        LOG.debug("Creating swapfile in '%s' on fstype '%s' using '%s'",
+                  fname, fstype, method)
+
+        if method == "fallocate":
+            cmd = ['fallocate', '-l', '%dM' % size, fname]
+        elif method == "dd":
+            cmd = ['dd', 'if=/dev/zero', 'of=%s' % fname, 'bs=1M',
+                   'count=%d' % size]
+
+        try:
+            util.subp(cmd, capture=True)
+        except util.ProcessExecutionError as e:
+            LOG.warning(errmsg, fname, size, method, e)
+
+            if method == "fallocate":
+                LOG.warning("Will attempt with dd.")
+                create_swap(fname, size, "dd")
+            else:
+                unlink_f(fname)
+
+    tdir = os.path.dirname(fname)
```

Comment:
What does the `t` in `tdir` here stand for?  (My brain interpreted it, incorrectly, as "temp". Is there a more descriptive name we could use?)

Reply:
This name was inherited from the precious version pre-refactoring :-) but yeah, I agree. Perhaps the meaning is temporary, but it's not correct. I'll just change it.

Reply:
Oh I see now. This name is also being used on the function setup_swap(). So the first author of this patch just wanted to keep the same naming scheme, perhaps?

Reply:
I mostly like to keep commits and changes to the smallest related to the topic. But I'm gonna change setup_swap() as well.